src/miscellaneous.py

```diff
@@ -0,0 +1,154 @@
+import sys, json
+
+sys.path.append("..")
+from app import *
+from src.util import *
+from flask import Flask, abort, request, jsonify, render_template
+from flask_restful import reqparse, abort, Api, Resource
+from bson.json_util import dumps, loads
+
+parser = reqparse.RequestParser()
+parser.add_argument("tag")
+parser.add_argument("display")
+
+
+class Hashtag(Resource):
+    def get(self):
+        """Gets the hastag list.
+
+        Returns:
+            A list of hashtag details
+
+        Example:
+        [
+            {
+                "name": "example_tag_1",
+                "is_display": 1
+            },
+            {
+                "name": "example_tag_2",
+                "is_display": 1
+            },
+            {
+                "name": "example_tag_3",
+                "is_display": 0
+            }
+        ]
+        """
+        request.get_json()
+        args = parser.parse_args()
+        print(args)
+        if args["display"] and args["tag"]:
+            abort_invalid_input(
+                "Invalid Input(400): Please check you input parameters!"
+            )
+        db = get_db(db_name="miscellaneous")
+        coll = db["hashtag"]
+        if args["tag"]:
+            cursor = coll.find({"name": str(args["tag"])})
+        elif args["display"]:
+            cursor = coll.find({"is_display": str(args["display"])})
+        else:
+            cursor = coll.find()
+        res = []
+        for tag in cursor:
+            res.append({"name": tag["name"], "is_display": tag["is_display"]})
+        return res
+
+    def post(self):
+        """Adds a tag into hashtag list.
+
+        Args(from request body):
+            tag: tag name
+            display: display status (default 1)
+        Returns:
+            Post result or error
+        """
+        request.get_json()
+        args = parser.parse_args()
+        try:
+            tag = str(args["tag"])
+            if "display" not in args:
```

Comment:
should be: if args['display'] is None:

Please try test case: http://127.0.0.1:5000/hashtag?tag=guo_test with POST